clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/converters/AbstractCloudFoundryAtomicOperationConverter.java

```diff
@@ -40,9 +41,18 @@
     CloudFoundrySpace space = CloudFoundrySpace.fromRegion(region);
 
     // fully populates the space guid which is what Cloud Foundry's API expects as an input, not the name.
-    return client.getOrganizations()
+    Optional<CloudFoundrySpace> spaceOptional = client.getOrganizations()
       .findByName(space.getOrganization().getName())
       .map(org -> client.getSpaces().findByName(org.getId(), space.getName()));
+
+    spaceOptional.ifPresent(spaceCase -> {
+      String[] parts = region.split(">");
```

Comment:
The region split and trim here are duplicate what is already done in fromRegion above. This is checking more than just case and seems overly complicated.

Reply:
Yes, you're right. Let me take a stab at it.